modules/varnish/templates/default.vcl.erb

```diff
@@ -184,8 +184,13 @@ sub vcl_fetch {
 }
 
 sub vcl_hash {
-  hash_data(req.url);
-  return(hash);
+    hash_data(req.url);
+    if (req.http.host) {
+        hash_data(req.http.host);
+    } else {
+        hash_data(server.ip);
+    }
+    return (lookup);
 }
```

Comment:
I think `return (lookup)` might be for a later version of Varnish (4) whereas we're on 3.0.5 and vcl_hash expects `return(hash)`.

I do wonder if we even need to define a vcl_hash at all, since I believe we're just resetting this to the default one (although it is painful trying to find what it does by default) noting that we also don't define a `vcl_pass` function in a similar way.

Reply:
Oops. And yes, I originally thought it better to be explicit, but that's unhelpful unless we do it consistently. I've referred to the docs, which do show the default behaviour as your described.

Could you take another look?